rust/agama-network/src/nm/dbus.rs

```diff
@@ -125,9 +130,13 @@ pub fn connection_to_dbus<'a>(
                     ("mtu", Value::new(conn.mtu)),
                     (
                         "assigned-mac-address",
-                        Value::new(conn.mac_address.to_string()),
+                        Value::new(conn.custom_mac_address.to_string()),
                     ),
                 ]));
+
+                if let Some(mac) = conn.mac_address {
+                    wireless_dbus_key.insert("mac-address", Value::new(mac.as_bytes()));
```

Comment:
NP: the name `wireless_dbus_key` is a bit confusing as I kind of expect something else.